modules/porous_flow/test/tests/sinks/PorousFlowPiecewiseLinearSink_BC_eg1.i

```diff
@@ -0,0 +1,131 @@
+## This is an example input file showing how to set a Type I BC with PorousFlowPiecewiseLinearSink
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 10
+  xmin = 0
+  xmax = 1
+  ny = 10
+  ymin = 0
+  ymax = 1
+[]
+
+[MeshModifiers]
+  [./aquifer]
+    type = SubdomainBoundingBox
+    block_id = 11
+    bottom_left = '0 0 0'
+    top_right = '1 1 1'
+  [../]
+[]
+
+[GlobalParams]
+  PorousFlowDictator = dictator
+[]
+
+[Variables]
+  [./porepressure]
+    initial_condition = 1.e6 # initial pressure in domain
+  [../]
+[]
+
+[PorousFlowBasicTHM]
+  porepressure = porepressure
+  coupling_type = Hydro
+  gravity = '0 0 0'
+  fp = the_simple_fluid
+  number_fluid_phases = 1
+  number_fluid_components = 1
+[]
+
+[AuxVariables]
+  [./fluxes_out]
+  [../]
+[]
+
+[BCs]
+  [./in_left]
+    type = PorousFlowPiecewiseLinearSink
+    variable = porepressure
+    boundary = 'left'
+    pt_vals = '-1e9 1e9' # x coordinates defining g
+    multipliers = '-1e9 1e9' # y coordinates defining g
+    PT_shift = 2.E6   # BC pressure
+    flux_function = 1E-6 # Variable C
+    fluid_phase = 0
+    save_in = fluxes_out
+  [../]
+[]
+
+[Postprocessors]
```

Comment:
I suggest you make the "gold" file be the CSV with left_flux in it.  You should ensure that this is correctly calculated, either using the "Erf" function solution for this, or by numerically looking at the derivative and computing the flux, or by using PorousFlowFluidMass to compute the mass in the system and showing conservation of mass (probably the easiest way).

Also, the CSV file could contain a Postprocessor that's the PointValue of your pressure, to demonstrate that pressure is indeed fixed to 2E6

Reply:
Ok, the csv gold file shows the fluxes and pressures at the left and right boundary. They match the steady state analytical solution by the end of the numerical solution.